test/performance/mako/sidecar.go

```diff
@@ -16,9 +16,47 @@ limitations under the License.
 
 package mako
 
+import (
+	"context"
+	"fmt"
+
+	"knative.dev/pkg/changeset"
+	"knative.dev/pkg/injection/clients/kubeclient"
+	"github.com/golang/protobuf/proto"
+        "github.com/google/mako/helpers/go/quickstore"
+        qpb "github.com/google/mako/helpers/proto/quickstore/quickstore_go_proto"
+)
+
 const (
 	// SidecarAddress is the address of the Mako sidecar to which we locally
 	// write results, and it authenticates and publishes them to Mako after
 	// assorted preprocessing.
-	SidecarAddress = "localhost:9813"
+	sidecarAddress = "localhost:9813"
 )
+
+// SetupMako sets up the mako client for the provided benchmarkKey.
+// It will add a few common tags and allows each benchmark to add custm tags as well.
+// It returns the mako client handle to sotre metrics, a method to close the connection
+// to mako server once done and error if case of failures.
+func SetupMako(ctx context.Context, benchmarkKey string, extraTags []string)(*quickstore.Quickstore, func(context.Context), error) {
+	var close func(context.Context)
+	tags := []string{"master"}
+        if commitID, err := changeset.Get(); err == nil {
+                tags = append(tags, commitID)
+        } else {
+		return nil, close, err
+        }
+
+	// Get the Kubernetes version from the API server.
+	if version, err := kubeclient.Get(ctx).Discovery().ServerVersion(); err == nil {
+		tags = append(tags, fmt.Sprintf("kubernetes=%s", version))
+	} else {
+		return nil, close, err
+	}
+
+	tags = append(tags, extraTags...)
+        return quickstore.NewAtAddress(ctx, &qpb.QuickstoreInput{
+                BenchmarkKey: proto.String(benchmarkKey),
```

Comment:
As a follow-on change, I'd like to see us move the `benchmarkKey` into something we pull from `kodata` and symlinked `mako.config` files (similar to `HEAD` and changeset).  Can you add a TODO?